cplusplus/VImage.cpp

```diff
@@ -777,6 +777,28 @@ VImage::write_to_target( const char *suffix, VTarget target,
 			set( "target", target ) );
 }
 
+VImage 
+VImage::thumbnail_buffer( void *buf, size_t len, int width, VOption *options )
+{
+	VipsBlob *blob;
+
+	/* We don't take a copy of the data or free it.
+	 */
+	blob = vips_blob_new( NULL, buf, len );
+
+	try
+	{
+		VImage thumbnail = thumbnail_buffer( blob, width, options );
+		vips_area_unref( VIPS_AREA( blob ) );
+		return( thumbnail );
+	}
+	catch ( VError &ex )
+	{
+		vips_area_unref( VIPS_AREA( blob ) );
+		throw; // rethrow exception
+	}
```

Comment:
I think this try-catch can be avoided like this:
```suggestion
	VImage out;

	/* We don't take a copy of the data or free it.
	 */
	blob = vips_blob_new( NULL, buf, len );
	options = (options ? options : VImage::option())->
		set( "buffer", blob )->
		set( "width", width )->
		set( "out", &out );
	vips_area_unref( VIPS_AREA( blob ) );

	call( "thumbnail_buffer", options );

	return( out );
```

See for example the implementation of `VImage::new_from_buffer`:
https://github.com/libvips/libvips/blob/80f1d89365ecd69b268e7ffd22fed8b0d4f5fa62/cplusplus/VImage.cpp#L596-L606